### ISSUE TITLE
[MNG-5224] REGRESSION: Injected Settings in a Mojo are missing the profiles from settings.xml

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#6985]
**[Stephen Connolly](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=stephenc)** opened **[MNG-5224](https://issues.apache.org/jira/browse/MNG-5224?redirect=false)** and commented

If you have a Mojo with

```
/**
 * @parameter expression="${settings}"
 * @required
 * @readonly
 */
private Settings settings;
```

In Maven 2.2.1, the Settings included all the profiles, in Maven 3.0.3 the profiles are all missing.

This issue becomes critical, as most users using Maven Release Plugin will be storing their gpg.passphrase as a property in a profile in their settings.xml, and as of Maven Release Plugin 2.2.2 we serialize the injected Settings to a file with the result that most users will be unable to use Maven 3.0.3 with Maven Release Plugin 2.2.2

---

**Affects:** 3.0.3

**Issue Links:**
- [MRELEASE-724](https://issues.apache.org/jira/browse/MRELEASE-724) Release plugin doesn't honor profiles defined in user settings
 (_**"is duplicated by"**_)
- [MNG-5036](https://issues.apache.org/jira/browse/MNG-5036) Settings.getProfiles() returning an empty list
 (_**"is duplicated by"**_)
- [MRELEASE-577](https://issues.apache.org/jira/browse/MRELEASE-577) release:prepare does not pass argument "--settings" with current settings.xml to inner maven

- [MRELEASE-724](https://issues.apache.org/jira/browse/MRELEASE-724) Release plugin doesn't honor profiles defined in user settings

- [MRELEASE-756](https://issues.apache.org/jira/browse/MRELEASE-756) release:prepare ignores properties from settings.xml

**Comment by jira-importer** [comment]
**[Stephen Connolly](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=stephenc)** commented

r1226842 contains the fix... needs test case before I am willing to mark this as closed

**Comment by jira-importer** [comment]
**[Olivier Lamy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=olamy)** commented

merged in 3.0.4RC branch
core it test added
